src/python/pants/python/python_setup.py

```diff
@@ -127,6 +127,19 @@ def register_options(cls, register):
                 "Mutually exclusive with `[python-setup].requirement_constraints`."
             ),
         )
+        register(
+            "--experimental-resolves-to-lockfiles",
```

Comment:
I debated calling this only `--experimental-resolves`. But I think there's value in including the word "lockfile" in the name. "Resolve" on its own is vague to most users.

Reply:
IMO, just `resolves`. The name enough will never be enough to skip reading the help, and the help should say that bit very clearly.

Reply:
Are you okay with waiting a little longer for this rename? https://github.com/pantsbuild/pants/issues/12714 is making me think we _might_ want >1 resolve related option, e.g. `default_resolve` to be a dedicated option. Unclear.

I'm open to renaming but would like to keep this for now while iterating. It helps me my mental model.